ValidateTree.java

```diff
@@ -0,0 +1,53 @@
+/*
+This approach solves the problem by setting a min and max value at each node, propogated from the previous node,
+and then checking that the current value of the node is within the range of that minimum and maximum. This approach 
+is an iterative implementation, where the recursion stack is manually handled.
+
+Did it run on Leetcode: Yes
+
+Problems with this code: The code is failing at test case [Integer.MIN_VALUE, Integer.MIN_VALUE]
+
+*/
+class Solution {
+    //Time Complexity = O(n)
+    //Space Complexity = O(h)
+    public boolean isValidBST(TreeNode root) {
+        
+        //We have three different stacks to store the root, min and max value
+        Stack<TreeNode> currStack = new Stack<>();
+        Stack<Long> minStack = new Stack<>();
+        Stack<Long> maxStack = new Stack<>();
+        
+        //Long datatype is used because the range of values is from MAX int to MIN int
+        long min = Integer.MIN_VALUE - 1;
+        long max = Integer.MAX_VALUE + 1;
+        
```

Comment:
Here, the min having value ( === Integer.MAX_VALUE ) and max having value (==== Integer.MIN_VALUE). So, check conditions again(There I used swapped values for your example and it gave me correct answer),  then let me know what happens or also post your solution in your cohort so everyone can see your approach as it is different and new.